docs/proposals/resource_management.md

```diff
@@ -0,0 +1,313 @@
+metadata:
+    name: Pod3
+spec:
+    containers:
+        name: foo
+            resources:
+                limits:
+                    cpu: 50m
+                    memory: 2Gi
+                requests:
+                    cpu: 20m
+                    memory: 1Gi
+        name: bar
+            resources:
+                limits:
+                    cpu: 100m
+                        memory: 1Gi
+```
+
+```yaml
+kind: Pod
+metadata:
+    name: Pod4
+spec:
+    containers:
+        name: foo
+            resources:
+                limits:
+                    cpu: 20m
+                    memory: 2Gi
+                requests:
+                    cpu: 10m
+                    memory: 1Gi  
+```
+
+Pod3 and Pod4 are both classified as Burstable and are hence nested under the Bu cgroup <br>
+/Bu/cpu.shares = 130m <br>
+/Bu/Pod3/cpu.quota = 150m <br>
+/Bu/Pod4/cpu.quota = 20m <br>
+/Bu/m.limit_in_bytes = Allocatable - 5Gi <br>
+/Bu/Pod3/m.limit = 3Gi <br>
+/Bu/Pod4/m.limit = 2Gi <br>
+
+#### Best Effort Cgroup ####
+
+For pods belonging to the Best effort Qos we don't specify any cpu parameters for the pod cgroup.<br>
+
+/BE/cpu.shares = 2 <br>
+/BE/cpu.quota= not set
+/BE/m.limit_in_bytes = Allocatable - summation of memory requests/limits of guaranteed pods <br>
+/BE/PodUID/m.limit = no limit <br>
+
+Example:
+We have a pod 'Pod5' having Pod Spec given below:
+
+```yaml
+kind: Pod
+metadata:
+    name: Pod5
+spec:
+    containers:
+        name: foo
+            resources:
+        name: bar
+            resources:
+```
+
+Pod5 is classified as Best effort and is hence nested under the BE cgroup
+
+/BE/cpu.shares = 2 <br>
+/BE/cpu.quota= not set <br>
+/BE/m.limit_in_bytes = Allocatable - 5Gi <br>
+/BE/Pod5/m.limit = no limit <br>
+
+### With Systemd
+
+In systemd we have slices for the three top level QoS class. Further each pod is a subslice of exactly one of the three QoS slices. Each container in a pod belongs to a scope nested under the qosclass-pod slice.
+
+Example:<br>
+We plan to have the following cgroup hierarchy on systemd systems
+
+```
+/memory/G.slice/G-PodUID.slice/containerUID.scope
+/cpu,cpuacct/G.slice/G-PodUID.slice/containerUID.scope
+/memory/Bu.slice/Bu-PodUID.slice/containerUID.scope
+/cpu,cpuacct/Bu.slice/Bu-PodUID.slice/containerUID.scope
+/memory/BE.slice/BE-PodUID.slice/containerUID.scope
+/cpu,cpuacct/BE.slice/BE-PodUID.slice/containerUID.scope
+```
+
+There are similar slices for Burstable and Best Effort QoS classes.
+
+## Implementation Plan
+
+The implementation plan is outlined in the next sections.
+
+#### Top level Cgroups for QoS tiers
+
+Three top level cgroups for Guaranteed, Burstable and Best effort Qos classes are created when Kubelet starts to run on a node. We use libcontainers cgroups manager interface to create/destroy the same. These cgroups are only created once on Kubelet initialization as a part of node setup. The Guranteed pods Cgroup would be nested directly under the root by default.
```

Comment:
Yeah right! We don't have support for slice management in systemd yet. So we still have to figure out on how to add support for the same. 

But for raw cgroup fs we can pass out a fake pid (0 in this case) to `Apply()` for creating cgroups. And in general as far I have seen it has support for basic cgroup management such as destruction, update etc. Please correct me if I am wrong.

But yeah having a something which can work in the absence of a pid would be the best way to go. I will discuss offline with @vishh and get back.